### PR TITLE
Uk companies

### PR DESCRIPTION
**Pull request by IegorSoboliev**

### CODE REVIEW

File: business_register/converter/uk_company.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+# Standard instance of a logger with __name__
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+class Uk_Company_Converter(Ukr_CompanyConverter):
+
+    def __init__(self):
+        super().__init__()
```
Comment (roman-tiukh) [inline business_register/converter/uk_company.py:17]: redundant

File: business_register/converter/uk_company.py [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+                        company.status = status
+                        update_fields.append('status')
+                    if convert_date_to_string(company.registration_date) != registration_date:
+                        company.registration_date = registration_date
+                        update_fields.append('registration_date')
+                    if company.code != code:
+                        company.code = code
+                        update_fields.append('code')
+
+                    if len(update_fields):
```
Comment (roman-tiukh) [inline business_register/converter/uk_company.py:78]:
```suggestion
                    if update_fields:
```

File: business_register/models/company_models.py [hunk 3]
```diff
@@ -63,7 +66,6 @@ def is_closed(self):
 
     class Meta:
         verbose_name = 'компанія/організація'
-        ordering = ['id']
```
Comment (roman-tiukh) [inline business_register/models/company_models.py:66]: why you delete this??

**Review by roman-tiukh — CHANGES_REQUESTED** [review]

File: business_register/converter/uk_company.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+from business_register.models.company_models import Company
+from data_ocean.utils import format_date_to_yymmdd, convert_date_to_string
+from location_register.converter.address import AddressConverter
+
+# Standard instance of a logger with __name__
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+
+
+class Uk_Company_Converter(Ukr_CompanyConverter):
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/uk_company.py:14]: Sorry))). Fixed

File: business_register/converter/ukr_company.py [hunk 5]
```diff
@@ -14,10 +14,11 @@
     Signer, TerminationStarted
 )
 from data_ocean.converter import BulkCreateManager
-from data_ocean.utils import cut_first_word, format_date_to_yymmdd, get_first_word
+from data_ocean.utils import (cut_first_word, format_date_to_yymmdd, get_first_word,
+                              convert_date_to_string)
 
 
-class CompanyConverter(BusinessConverter):
+class Ukr_CompanyConverter(BusinessConverter):
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/ukr_company.py:21]: Sorry again))). Fixed

File: data_ocean/utils.py [hunk 6]
```diff
@@ -53,5 +53,15 @@ def cut_first_word(string):
...
+    if '/' in str_ddmmyy:
+        str_ddmmyy = str_ddmmyy.replace("/", ".")
+    return datetime.datetime.strptime(str_ddmmyy, "%d.%m.%Y").strftime("%Y-%m-%d")
+
+
+# checking if date exists, then converting it to string
+def convert_date_to_string(date):
+    if date:
+        return str(date)
+    return None
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply data_ocean/utils.py:67]: renamed

File: business_register/converter/uk_company.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                company = Company.objects.filter(edrpou=number).first()
+                if not company:
+                    company = Company(name=name,
+                                      company_type=company_type,
+                                      edrpou=number,
+                                      address=address,
+                                      country=country,
+                                      status=status,
+                                      registration_date=registration_date,
+                                      code=code)
```
[the remark this answers is not in the document]
Comment (IegorSoboliev) [reply business_register/converter/uk_company.py:52]: done

File: business_register/converter/uk_company.py (hunk 2 above)
Comment (IegorSoboliev) [reply business_register/converter/uk_company.py:78]: done

File: business_register/models/company_models.py (hunk 3 above)
Comment (IegorSoboliev) [reply business_register/models/company_models.py:66]: we do have such ordering in the parent Model

Comment (roman-tiukh) [reply business_register/models/company_models.py:66]: you redefined the meta class and all the meta properties of the parent class were overwritten

Comment (roman-tiukh) [reply business_register/models/company_models.py:66]: meta properties are not inherited if you write a meta class in a child class

Comment (IegorSoboliev) [reply business_register/models/company_models.py]: restored, thank you

**Review by roman-tiukh — APPROVED** [review]

**Pull request merged**